--- colour-0.3.8/colour/plotting/diagrams.py	2016-07-09 09:02:44.000000000 +0200
+++ diagrams.py	2017-12-05 23:09:43.211533906 +0100
@@ -30,6 +30,7 @@
 from colour.algebra import normalise_vector
 from colour.colorimetry import spectral_to_XYZ
 from colour.models import (
+    Lab_to_XYZ,
     Luv_to_uv,
     Luv_uv_to_xy,
     UCS_to_uv,
@@ -502,6 +503,7 @@
 
 
 def CIE_1976_UCS_chromaticity_diagram_plot(
+        Lab_array,
         cmfs='CIE 1931 2 Degree Standard Observer',
         show_diagram_colours=True,
         **kwargs):
@@ -559,6 +561,11 @@
                color='black',
                linewidth=2)
 
+    # plot Lab color
+    for Lab in Lab_array: 
+        Lab_uv = Luv_to_uv(XYZ_to_Luv(Lab_to_XYZ(Lab,illuminant),illuminant),illuminant)
+       	pylab.plot(Lab_uv[0],Lab_uv[1],'+',color='black',markeredgewidth=1,markeredgecolor='black',markerfacecolor='none')
+
     for label in labels:
         u, v = wavelengths_chromaticity_coordinates.get(label)
         pylab.plot(u, v, 'o', color='black', linewidth=2)
@@ -607,7 +614,7 @@
         'x_label': 'CIE u\'',
         'y_label': 'CIE v\'',
         'grid': True,
-        'bounding_box': (0, 1, 0, 1)})
+        'bounding_box': (0.1, 0.4, 0.4, 0.7)})
     settings.update(kwargs)
 
     boundaries(**settings)
